content/blog/2019-09-25-RecordTypes.md

```diff
@@ -95,29 +95,73 @@ This syntax was designed to have a similar format as record instantiation.
 Immutable data structures are easier to reason about when developing compiler optimizations. However, forcing developers to recreate the entire record every time they need to change a value poses usability challenges. We decided on sticking to immutability in part for optimizations' sake. The ability to make guarantees that a value will not change lends itself well to constant-folding, even after a function call that takes in your record as an argument. 
 
 ### Evaluation
-We set out to implement record types and we successfully implemented immutable nominal record types. Record types do not have any restrictions except that two fields cannot have the same name and that a record type cannot be recursive. Besides that, everything is allowed. Record types play nicely with existing types and new syntactical additions follow previous precedents set by Bril or otherwise follow precedents set by other languages that have record types. Record declaration, instantiation, and access are all (we think) intuitive and straightforward, and this means our record types provide good value as an addition to the Bril language. 
+To evaluate our implementation, we need to check that our implementation works as specified as a language feature, and that this language features satisfies the goal of extending Bril's ability to compile higher-level languages with record data types.
 
+One of the goals of record types was to allow Bril to logically group and use related data points. This feature is useful when compiling higher-level languages that utilize a similar data structure like records in OCaml or structs in C into Bril. 
+
+To measure success of this goal, we considered whether the functionality added from these new types increases Bril's ability to compile higher-level languages with record type data structures.
+
+We implemented immutable nominal record types. This implementation includes record declaration, instantiation, and access. Decisions about which operations to include in our design were influenced by the operations available on record data types in higher-level languages. To test this functionality, we created a suite of tests that covered each of these operations, as well as combinations of these operations.
+
+The primary operation not supported by our record type specification that is supported by some higher-level language record types is mutation. While our record types do not support mutation as discussed, this does not significantly hinder Bril's ability to compile higher-level languages that support this feature. 
+
+Consider the following C code that declares a struct and provides a method to update.
+```
+struct Person {
+   int age;
+   bool isAsleep;
+};
+
+void updateAge(Person p){
+    p->age = p->age + 1;
+}
+
+int main(void) {
+    Person Henry = { 20, false };
+    updateAge(Henry);
```

Comment:
```suggestion
    Henry.age += 1;
```
Can we inline the function to make it more similar to the bril?

Reply:
yea for sure!